testng-core/src/main/java/org/testng/SuiteRunner.java

```diff
@@ -254,7 +252,7 @@ public void setReportResults(boolean reportResults) {
   }
 
   private void invokeListeners(boolean start) {
-    for (ISuiteListener sl : listeners.values()) {
+    for (ISuiteListener sl : Lists.newArrayList(listeners.values())) {
```

Comment:
Using a new ArrayList here won't change things

Reply:
Yes they would.  This was done to accommodate the tests that are part of https://github.com/cbeust/testng/issues/2061

Reply:
How is it possible? The order of the list will depend on the order of `listeners.values()` and I can't remember `Lists.newArrayList` make a sort.

Reply:
@juherr  its insertion order remember ? So the order in which `listener.values()` provides is the way in which they would get added to the new list as well.

Reply:
I think `listeners.values()` and `Lists.newArrayList(listeners.values())` will both produce the same same Iterator.